expat/lib/xmlparse.c

```diff
@@ -2544,13 +2557,15 @@ XML_GetFeatureList(void) {
      XML_L("XML_BLAP_ACT_THRES"),
      EXPAT_BILLION_LAUGHS_ATTACK_PROTECTION_ACTIVATION_THRESHOLD_DEFAULT},
 #endif
+    /* Added in Expat 2.6.0. */
+    {XML_FEATURE_DTD, XML_L("XML_GE"), XML_GE * 2 - 1},
```

Comment:
Why the `*2-1` expression?

Reply:
I started out with plain `XML_GE` to give 0 or 1 but value 0 will show as misleading `[..], XML_GE, [..]` in `xmlwf -v` output then and only serving feature `XML_GE` in the list when `XML_GE == 1` would take away the ability to distinguish Expat prior to introduction of `XML_GE` and Expat with `XML_GE` deliberately disabled. `XML_GE * 2 - 1` gives 1 or -1 where -1 was intended to signal disabled. With that said, what do you think of it now?

Reply:
I think I would prefer only serving XML_GE when it's 1 -- that's consistent with how all the other features work.

XML_GE==0 will most likely be used for special applications that control Expat themselves, and therefore won't check the feature list anyway.

If you _really_ feel a need to make <2.6.0 distinguishable from >=2.6.0&&GE==0, maybe serve a XML_GE_DISABLED feature instead? That'd make some sense, actually:

* XML_GE_DISABLED present --> the new variant of Expat -- `Expat >= 2.6.0 && XML_GE==0`.
* XML_GE_DISABLED _not_ present --> the old variant of Expat -- `Expat < 2.6.0` _or_ `Expat >= 2.6.0 && XML_GE==1`

Reply:
@Snild-Sony I feel we have a bit of a pest or cholera situation here: inconsistency flavor one (1/-1) versus inconsistency flavor two (negated feature). I notice that case `Expat >=2.6.0 && XML_GE==0` could always be detected reliably at runtime by asking Expat parse a tiny piece of XML and see whether an entity was served as a self-reference; wherever that turns out really needed. Maybe then approach "only serve feature XML_GE when enabled" — your initial recommendation — matches credo "simple/common things easy and complex/rare things possible" in practice then, which is maybe not too bad. Pushing that new simple approach now…

Reply:
Sounds reasonable to me.